services/QuillLMS/client/app/bundles/Teacher/components/premium/school_and_district_premium_modal.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+
+import SchoolSelector, { NOT_LISTED, } from '../shared/school_selector'
+import { requestPost, requestPut, } from '../../../../modules/request';
+import { closeIcon, DropdownInput, Spinner, } from '../../../Shared/index'
+
+export const SCHOOL_PURCHASE_STAGE_TWO = 'school_purchase_stage_two'
+
+const SchoolAndDistrictPremiumModal = ({ stripeSchoolPlan, associatedSchools, userIsSignedIn, startAtStageTwo, customerEmail, closeModal, handleNotListedSelection, }) => {
+  const [stage, setStage] = React.useState(startAtStageTwo ? 2 : 1)
+  const [selectedSchool, setSelectedSchool] = React.useState(null)
+  const [loading, setLoading] = React.useState(false)
+
+  React.useEffect(handleStageTwo, [])
+  React.useEffect(handleStageTwo, [stage])
+
+  function goToStage2() {
+    setLoading(true)
+    setStage(2)
+  }
+
+  function handleStageTwo() {
+    if (stage !== 2) { return }
+
+    if (associatedSchools.length === 1) {
+      const schoolIds = associatedSchools.map(s => s.id)
+      goToStripe(schoolIds)
+    } else {
+      setLoading(false)
+    }
+  }
+
+  function goToStripe(schoolIds) {
+    const path = '/stripe_integration/subscription_checkout_sessions'
+    const data = {
+      cancel_path: 'premium',
+      customer_email: customerEmail,
+      ...(schoolIds && { school_ids: JSON.stringify(schoolIds) }),
+      stripe_price_id: stripeSchoolPlan.plan.stripe_price_id
+    }
+
+    requestPost(path, data, body => { window.location.replace(body.redirect_url) })
+  }
```

Comment:
It wasn't possible to reuse the existing button component here, because we need to be able to activate this logic indirectly (like when a user with one school is signed out, pressing "Buy Now" needs to redirect them to sign in and then send them directly to Stripe, or when a user with no schools selects one from the school selection component there isn't an additional button to press).